### ISSUE TITLE
{{deps}} moustaches should work in runtime.env: keys

### ISSUE THREAD

**Issue by jhheider** [teaxyz/cli#497]
https://github.com/teaxyz/pantry/actions/runs/4581371758/jobs/8090857190

For now, I'll just export this from ca-certs, so git will work.

**Comment by mxcl** [comment]
There's plenty of places where the `{{moustaches}}` aren't fully hydrated. It mostly depends on what data the function has to hand.

I think here the information about what dependencies a specific package has is lost. However we could just pass all the packages in the fully hydrated dependency graph. It's more than needed but that does include the `deps`.

**Comment by jhheider** [comment]
Since we don't hydrate multiple versions currently (right? just conflict them out?), all deps should have a unique identity for a given dep.

Possibly we could hydrate this envvar using $SSL_CERT_FILE, but we'd have to know that would populate first...

**Comment by mxcl** [comment]
> Since we don't hydrate multiple versions currently (right?)

Yes this is correct. So far we haven't needed more granularity than that.

> Possibly we could hydrate this envvar using $SSL_CERT_FILE, but we'd have to know that would populate first...

I added `env.FOO` to the moustaches recently so that *should* work. Though I recall humming and harring about whether this was a good idea. So maybe I didn't and only added `env.HOME`.

**Comment by jhheider** [comment]
It seems versatile, but we should aspire to never need arbitrary `env` vars. Unless/until we want optional build features; then we're recreating `emerge`.

**Comment by mxcl** [comment]
indeed. This was my reasoning and you spelling it out makes me think that yeah it's a bad call. Let me go check what I did.

**Comment by jhheider** [comment]
`teaxyz/teash`

**Comment by jhheider** [comment]
That sounds like the use case I needed. Transitive sounds like you'd be doing it wrong anyway; should be explicit.

**Comment by mxcl** [comment]
my bad, I was misreading the output, let me see why it doesn't work
